tests/test_update.py

```diff
@@ -195,3 +196,32 @@ def test_merge_deprecation() -> None:
     with warns(UserWarning, match=re.escape(msg)):
         OmegaConf.update(cfg, "a", {"c": 20})  # default to set, and issue a warning.
         assert cfg == {"a": {"c": 20}}
+
+
+@mark.parametrize(
+    "cfg,key,value,expected",
+    [
+        param({}, "a", 10, {"a": 10}, id="add_value"),
+        param({}, "a.b", 10, {"a": {"b": 10}}, id="add_value"),
+        param({}, "a", {"b": 10}, {"a": {"b": 10}}, id="add_dict"),
+        param({}, "a.b", {"c": 10}, {"a": {"b": {"c": 10}}}, id="add_dict"),
+        param({}, "a", [1, 2], {"a": [1, 2]}, id="add_list"),
+        param({}, "a.b", [1, 2], {"a": {"b": [1, 2]}}, id="add_list"),
+        param(
+            {"user": User(name="Bond", age=7)},
```

Comment:
This was how I initially attempted to implement it.
the downside is that flags are recursive, and in this case you want the flag to apply only to the structured config node itself and not to any nested dictionaries.

In general, Structured Configs works without the struct flag.
The struct flag can be used to open a Structured Config to changes if it's to False on an SC node. Usually this is done temporarily.

Reply:
Yeah I thought about the recursive issue after posting this message. I guess it'd be possible for a structured config to explicitly set `struct=False` onto its unstructured nested dicts & lists... Another idea could be to use a different flag specifically for structured configs... but anyway, like I said, probably not worth changing at this point. As long as the user-facing behavior is sensible, it's not a big deal.

Reply:
Recursively setting a flag on child nodes will have unintended consequences and in general very tricky in the presence of a dynamic object model.
as an example, if each child has an explicit struct flag set to False, it will be hard to set a specific node to struct mode (now we will also need to iterate on all the children to set the flag to True on them).

The problem with Structured Configs is that the the expected behavior is not recursive, not that they  are sharing the struct flag. They are in fact not using the struct flag, EXCEPT to temporarily remove the structness from an SC node.

If we ever need a second non-recursive flag we can consider adding a generic mechanism for it and port the behavior of the Structured Configs to use it.

Reply:
Right -- so maybe what we need is a mechanism to set a flag in a non-recursive way 🤔 (i.e., children wouldn't inherit from it)

Reply:
Something to consider for another PR :)